### PR TITLE
choose box from map

### PR DESCRIPTION
**Pull request by felixerdy**
This PR adds the function to choose a box from a map which is more convenient than typing in the senseBox id. 

As I used the [Mapbox Android SDK](https://www.mapbox.com/android-docs/map-sdk/overview/) you will need to paste a Mapbox token to `app/src/main/res/values/strings.xml`. You can get access tokens [here](https://www.mapbox.com/studio/account/tokens/) after creating a free Mapbox account.

As this is my first time I worked with Kotlin, please feel free to suggest improvements!

screenshot_20171005-153846

### CODE REVIEW

**Comment by dbaelz** [comment]
Thanks for the  pull request.

You're right, choosing the senseBox with the box id is really inconvenient. I planned to add a `search by name` feature, but the map is a great (and even better) idea.

I'll review the the pull request today or tomorrow. At first glance it looks good, but I have some suggestions and comments.

**Comment by dbaelz** [comment]
Please be aware: I changed the target branch to develop.

Open Sense uses (more or less) gitflow and all work for a new release should be merged into develop. The readme will be adjusted to give that hint to contributor.

**Comment by felixerdy** [comment]
@dbaelz Thank you for your suggestions, I did some improvements based on them.

File: app/src/main/java/de/codefor/karlsruhe/opensense/widget/base/BaseWidgetConfigurationActivity.kt [hunk 1]
```diff
@@ -31,31 +39,57 @@ abstract class BaseWidgetConfigurationActivity : AppCompatActivity() {
...
+        mapView = findViewById<View>(R.id.mapView) as MapView
+        mapView.onCreate(icicle)
+
+        mapView.getMapAsync { mapboxMap ->
+            run {
+                this.mapboxMap = mapboxMap
+
+                // get all boxes from api and display them on the map
+                WidgetHelper.getAllBoxes()
+                        .subscribe(this::displayBoxesOnMap)
```
Comment (felixerdy) [inline app/src/main/java/de/codefor/karlsruhe/opensense/widget/base/BaseWidgetConfigurationActivity.kt]: I was not able to figure out how to pass parameters to the subscription method, therefore I created a variable containing the `mapboxMap`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -20,4 +23,7 @@
     <!-- One Value Widget -->
     <string name="one_value_image_description">Cloud symbol</string>
     <string name="one_value_error_text">Error</string>
+
+    <string name="mapbox_token">## enter your mapbox token here ##</string>
```
Comment (felixerdy) [inline app/src/main/res/values/strings.xml]: What is the most elegant way to store security tokens of Android projects on GitHub?

Comment (dbaelz) [reply app/src/main/res/values/strings.xml]: In my opinion, the most elegant way is to store it token in the local gradle properties and include the stored value in the build process. The required changes are documented in #5. I start with this issue as soon as this pull request is merged.

File: app/src/main/java/de/codefor/karlsruhe/opensense/widget/base/BaseWidgetConfigurationActivity.kt (hunk 1 above)
Comment (dbaelz) [reply app/src/main/java/de/codefor/karlsruhe/opensense/widget/base/BaseWidgetConfigurationActivity.kt]: You could do it this way: `WidgetHelper.getAllBoxes().subscribe({boxes -> displayBoxesOnMap(mapboxMap, boxes)})` and add the MapboxMap argument to the method. But don't trouble yourself. If you like, I'll change it later on.

**Comment by dbaelz** [comment]
I merged your pull request and changed the open points in #5. Thanks for the great work!

**Pull request merged**